webserver/cWebem.cpp

```diff
@@ -1157,6 +1157,115 @@ namespace http {
+			// "X-Real-IP"	The (old) default header used by NGINX  (http://nginx.org/en/docs/http/ngx_http_realip_module.html#real_ip_header)
+			//
+			// These headers can occur multiple times, so need to be 'squashed' together
+			// And a single line can contain multiple (comma separated) values in order
+
+			bool bStatus = true;
+			std::vector<std::string> headers;
+			std::vector<std::string> hosts;
+
+			if (sumProxyHeader("Forwarded", req, headers))
+			{	// We found one or more Forwarded headers that need to be processed into a list of Hosts
+				if (!parseForwardedProxyHeader(headers, hosts))
+				{
+					bStatus = false;
+				}
+			}
+			else if (sumProxyHeader("X-Forwarded-For", req, headers))
+			{	// We found one or more X-Forwarded-For headers that need to be processed into a list of Hosts
+				if (!parseProxyHeader(headers, hosts))
+				{
+					bStatus = false;
+				}
+			}
+			else if (sumProxyHeader("X-Real-IP", req, headers))
+			{	// We found one or more X-Real-IP headers that need to be processed into a list of Hosts
+				if (!parseProxyHeader(headers, hosts))
+				{
+					bStatus = false;
+				}
+			}
+
+			if (hosts.size() >= 1)
+			{
+				realhost = hosts[0];	// Even if we found a chain of hosts, we always use the first (= origin)
+			}
+
+			return bStatus;
+		}
+
+		bool cWebem::sumProxyHeader(const std::string sHeader, const request &req, std::vector<std::string> &vHeaderLines)
+		{
+			uint8_t iFound = 0;
+
+			for (const auto &header : req.headers)
+			{
+				if (sHeader.compare(header.name) == 0)
+				{
+					iFound++;
+					vHeaderLines.push_back(header.value);
+				}
+			}
+
+			return (iFound > 0);
+		}
+
+		bool cWebem::parseProxyHeader(std::vector<std::string> &vHeaderLines, std::vector<std::string> &vHosts)
+		{
+			uint8_t iFound = 0;
+
+			for (const auto sLine : vHeaderLines)
+			{
+				std::vector<std::string> vLineParts;
+				StringSplit(sLine, ",", vLineParts);
+				for (std::string sPart : vLineParts)
+				{
+					iFound++;
+					stdstring_trimws(sPart);
+					vHosts.push_back(sPart);
+				}
+			}
+
+			return (iFound > 0);
+		}
+
+		bool cWebem::parseForwardedProxyHeader(std::vector<std::string> &vHeaderLines, std::vector<std::string> &vHosts)
+		{
+			uint8_t iFound = 0;
+
+			for (const auto sLine : vHeaderLines)
+			{
+				std::vector<std::string> vLineParts;
+				StringSplit(sLine, ",", vLineParts);
+				for (std::string sPart : vLineParts)
+				{
+					stdstring_trimws(sPart);
+					if (std::size_t isPos = sPart.find("for=") != std::string::npos)
+					{
+						iFound++;
+						isPos = isPos + 3;
+						std::size_t iePos = sLine.length();
+						if (sPart.find(";", isPos) != std::string::npos)
+						{
+							iePos = sPart.find(";", isPos);
+						}
+						std::string sSub = sPart.substr(isPos, (iePos - isPos));
+						stdstring_trimws(sSub);
+						vHosts.push_back(sSub);
+					}
+				}
+			}
+
+			return (iFound > 0);
```

Comment:
Instead of iFound we can use

`return !vHosts.empty();`

Reply:
see above

Reply:
There is no need to have a counting value in this function, speed wise, code size etc...